#Actividad M2.1 Uso de Google Colab
Emilio Alvarado González Aragón
A01704499

## 1. Prepara Google Colab para acceder a datos en Google Drive (mount)

Equipo 3

Juan Ramirez

Emilio Alvarado

Abraham Martinez

Fernanda Alanis

Leslie Ramos

Axl Lopez

Bernardo Cisneros

Tecnologico de Monterrey

In [1]:
import pandas as pd
import numpy as np
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [2]:
cd "/content/gdrive/My Drive/colab"

/content/gdrive/My Drive/colab


## 2. Guarda en un dataframe el contenido del dataset proporcionado

In [3]:
misdatos = pd.read_csv("titanic.csv")

## 3. Verifica que el archivo cargado en Google Colab corresponde con el archivo csv, mostrando  los primeros y los últimos renglones del "data frame" y comparándolos con los datos correspondientes en el archivo CSV.

In [4]:
misdatos.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
misdatos.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

La revisión de las primeras y últimas filas arroja igualdad con el archivo descargado de .csv, demostrando que se hizó la carga correcta del documento.




# 5. Descripción de Variables Agrega el encabezado Descripción de Variables

## 1. Identifica el tipo de variables presentes en el dataset, y en una celda describe cuáles variables son categóricas y cuáles numéricas. Muestra esta información en forma de lista

In [6]:
misdatos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


El uso de la función _info_, nos describe la siguiente composición:
- Variables numéricas: PassengerId, Pclass, Age, Sibsp, Parch, Fare
- Variables categóricas: Survived, Name, Sex, Ticket, Cabin, Embarked

## 2. Utiliza en diferentes celdas de código las funciones columns(), info() y describe() del dataframe. Muestra la información generada.

In [8]:
misdatos.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [10]:
misdatos.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [26]:
misdatos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## 3. Utiliza en diferentes celdas de código las funciones columns(), info() y describe() del dataframe. Muestra la información generada.

Las funciones anteriormente usadas, cumplen con el siguiente propósito:
- columns: enumera y despliega los títulos o etiquetas de las columnas del dataframe
- info: enumera cada columna con su conteo de valores no nulos y el tipo de dato de cada una
- describe: arroja ciertas estadísticas descriptivas de cada variable o columna, como su conteo, media, máximo, mínimo, desviación estándar y cuartiles

## 4. Indica cuántos registros se tienen en total, cuáles tienen valores nulos y cuáles no tienen valores nulos.

En total se tienen 891 registros

In [11]:
print(misdatos.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [13]:
print((misdatos.isnull()).sum().sum())

866


De los 891 registros, se tienen un total de 866 valores nulos. Estos valores nulos se encuentran mayormente en la variable de Age, Cabin y Embarked. 

## 5. ¿Qué columnas aparecen en el resultado de describe? ¿Cuál es la tarifa más cara? ¿Cuál es el promedio de edad?

Al usar el describe, obtenemos la visualización de las siguientes columnas: _PassengerId, Survived, Pclass, Age, Sibsp, Parch, Fare_

Esta misma función, nos permite observar que la tarifa mas cara es de 512.3292USD, y que el promedio de edad es de 29.29 años.

In [17]:
categorical_features = misdatos.loc[:, misdatos.dtypes == object].columns.values.tolist()
categorical_features.append("Survived")
print(categorical_features)

['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked', 'Survived']


## 6. Escribe el código necesario para indicar de las variables categóricas  cuantos valores diferentes se tiene por categoría. Puedes usar el método unique() para regresar los valores únicos de una columna.

In [21]:
for feature in categorical_features:
    print("Hay", len(misdatos[feature].unique()), "valores unicos en", feature)

Hay 891 valores unicos en Name
Hay 2 valores unicos en Sex
Hay 681 valores unicos en Ticket
Hay 148 valores unicos en Cabin
Hay 4 valores unicos en Embarked
Hay 2 valores unicos en Survived


# Análisis de datos

## 7. Agrega el código para indicar cuánto es el total que se pagó sumando la tarifa de todos los pasajeros

In [27]:
print(misdatos['Fare'].sum())

28693.9493


## 8. Crea un subconjunto de los datos de los que sobrevivieron y otro de los que no sobrevivieron usando df[condición]

In [22]:
survived = misdatos[misdatos['Survived']==1]
notSurvived = misdatos[misdatos['Survived']==0]

## 9. Mostrar los datos de 5 personas de las que sobrevivieron y de las que no sobrevivieron.

In [23]:
survived.head()

PassengerId  Survived  Pclass  \
1            2         1       1   
2            3         1       3   
3            4         1       1   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [25]:
notSurvived.head()

PassengerId  Survived  Pclass                            Name   Sex   Age  \
0            1         0       3         Braund, Mr. Owen Harris  male  22.0   
4            5         0       3        Allen, Mr. William Henry  male  35.0   
5            6         0       3                Moran, Mr. James  male   NaN   
6            7         0       1         McCarthy, Mr. Timothy J  male  54.0   
7            8         0       3  Palsson, Master. Gosta Leonard  male   2.0   

   SibSp  Parch     Ticket     Fare Cabin Embarked  
0      1      0  A/5 21171   7.2500   NaN        S  
4      0      0     373450   8.0500   NaN        S  
5      0      0     330877   8.4583   NaN        Q  
6      0      0      17463  51.8625   E46        S  
7      3      1     349909  21.0750   NaN        S

In [1]:
print("Hello World")
print("Hola mundo")
print("Bienvenidos al proyecto")
print("Ultimo cambio")

Hello World
Hola mundo
Bienvenidos al proyecto
Ultimo cambio
